### PR TITLE
feature_25363 : Deprecated maxBytesLocalHeap attribute in config

### PR DESCRIPTION
**Pull request by nishchay-naresh**

### CODE REVIEW

File: ehcache-core/src/main/java/net/sf/ehcache/config/CacheConfiguration.java [hunk 1]
```diff
@@ -392,12 +395,30 @@ public class CacheConfiguration implements Cloneable {
     private volatile Boolean copyOnWrite;
     private volatile boolean conflictingEternalValuesWarningLogged;
     private volatile Searchable searchable;
+    /**
+     * Cache Configuration attribute for size based cache installation
+     *
+     * @deprecated Use {@link #maxEntriesLocalHeap } instead
+     */
+    @Deprecated
```
Comment (jhouserizer) [inline ehcache-core/src/main/java/net/sf/ehcache/config/CacheConfiguration.java]: Not a problem, but a little odd to mark internal fields deprecated.

File: ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java [hunk 2]
```diff
@@ -194,7 +197,20 @@ private static void validateOverAllocation(RuntimeCfg config,  Long newValue) {
     private final Map<String, CacheConfiguration> cacheConfigurations = new ConcurrentHashMap<String, CacheConfiguration>();
     private ConfigurationSource configurationSource;
     private boolean dynamicConfig = DEFAULT_DYNAMIC_CONFIG;
+
+    /**
+     * Cache Configuration attribute for size based cache installation
+     *
+     * @deprecated Use {@link #maxBytesLocalOffHeap } instead
```
Comment (jhouserizer) [inline ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java]: Probably should suggest maxEntriesLocalHeap ?

File: ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java [hunk 3]
```diff
@@ -283,7 +299,10 @@ public boolean isMaxBytesLocalOffHeapSet() {
      * Checks whether the user explicitly set the maxBytesOnHeap
      * @return true if set by user, false otherwise
      * @see #setMaxBytesLocalHeap(Long)
+     * 
+     * @deprecated Use {@link #isMaxBytesLocalOffHeapSet() } instead
```
Comment (jhouserizer) [inline ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java:291]: isMaxBytesLocalOffHeapSet is not an "equivalent" or "alternative" to maxBytesLocalHeap ... it relates to the offheap tier...  they user will be wanting an alternative that relates to the heap tier

**Review by jhouserizer — CHANGES_REQUESTED** [review]

File: ehcache-core/src/main/java/net/sf/ehcache/config/CacheConfiguration.java (hunk 1 above)
Comment (nishchay-naresh) [reply ehcache-core/src/main/java/net/sf/ehcache/config/CacheConfiguration.java]: Checked it again, private attribute is not required to mark as @Deprecated, I have removed this.
(I have referred this protected attribute - overflowToDisk, which have been marked as @Deprecated)

File: ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java (hunk 3 above)
Comment (nishchay-naresh) [reply ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java:291]: This Configuration class consist configuration for CacheManager. On CacheManager level Heap tier pooling is available only be using maxBytesLocalHeap, maxEntriesLocalHeap is not there. So after maxBytesLocalHeap deprcation, next available pooling option is maxBytesLocalOffHeap which is at the Off-heap tier.

Comment (nnares) [reply ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java:303]: This Configuration class consist configuration for CacheManager. On CacheManager level Heap tier pooling is available only be using maxBytesLocalHeap, maxEntriesLocalHeap is not there. So after maxBytesLocalHeap deprcation, next available pooling option is maxBytesLocalOffHeap which is at the Off-heap tier.

**Comment by jhouserizer** [comment]
Hi @nnares  it's looking good. do you plan to squash your two commits or shall I just merge this?

**Review by jhouserizer — APPROVED** [review]

File: ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java (hunk 2 above)
Comment (nishchay-naresh) [reply ehcache-core/src/main/java/net/sf/ehcache/config/Configuration.java]: On CacheManager level Heap tier pooling is available only be using maxBytesLocalHeap, maxEntriesLocalHeap is not there. So after maxBytesLocalHeap deprcation, next available pooling option is maxBytesLocalOffHeap which is at the Off-heap tier.

**Comment by nishchay-naresh** [comment]
> Hi @nnares it's looking good. do you plan to squash your two commits or shall I just merge this?

Hi @jhouserizer squashed last two commits to one, Can merge the PR now.

**Pull request merged**